Cogs/Queue.py

```diff
@@ -0,0 +1,291 @@
+                               view=FreeChannelNotificationView(self, self.helper, self.queues[channel_type]["Entries"],
+                                                                game_number, queue_position))
+        else:
+            await self.announce_free_channel(game_number, queue_position + 1)
+
+    async def user_leave_queue(self, user: nextcord.Member):
+        self.queues["Regular"]["Entries"] = [entry for entry in self.queues["Regular"]["Entries"]
+                                             if entry["ST"] != user.id]
+        self.queues["Experimental"]["Entries"] = [entry for entry in self.queues["Experimental"]["Entries"]
+                                                  if entry["ST"] != user.id]
+        await self.update_queue_message(self.queues["Regular"])
+        await self.update_queue_message(self.queues["Experimental"])
+        with open(self.QueueLocation, "w") as f:
+            json.dump(self.queues, f)
+
+    @commands.command()
+    async def InitQueue(self, ctx: commands.Context, channel_type: ChannelTypeParameter):
+        if self.helper.authorize_mod_command(ctx.author):
+            await utility.start_processing(ctx)
+            channel_type = utility.get_channel_type(channel_type)
+
+            embed = nextcord.Embed(title=channel_type + " storytelling queue", description="Use >Enqueue to join")
+            if isinstance(ctx.channel, nextcord.Thread):
+                self.queues[channel_type]["ThreadId"] = ctx.channel.id
+                self.queues[channel_type]["ChannelId"] = ctx.channel.parent.id
+            elif isinstance(ctx.channel, nextcord.TextChannel):
+                # remove ThreadId in case the command was previously executed in a thread
+                self.queues[channel_type].pop("ThreadId", None)
+                self.queues[channel_type]["ChannelId"] = ctx.channel.id
+            else:
+                await utility.dm_user(ctx.author, 'Please place the queue in a text channel or thread')
+                return
+
+            queue_message = await ctx.send(embed=embed)
+            self.queues[channel_type]["MessageId"] = queue_message.id
+            self.queues[channel_type]["Entries"] = []
+
+            with open(self.QueueLocation, "w") as f:
+                json.dump(self.queues, f)
+            await self.helper.finish_processing(ctx)
+            print("-= The InitQueue command was used successfully by " + str(ctx.author.name) + " at " + str(
+                strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+        else:
+            await utility.deny_command(ctx, "InitQueue")
+            await utility.dm_user(ctx.author, "This command is restricted to moderators")
+        await self.helper.log(f"{ctx.author.mention} has run the InitQueue command in {ctx.channel.mention}")
+
+    @commands.command()
+    async def Enqueue(self, ctx: commands.Context, channel_type: ChannelTypeParameter, script: str, availability: str,
```

Comment:
There are some small plans for a text & live queue to work from the same Bot, so I'd be more inclined to have the command be something along the "TextQueueJoin", not sold on that being the best name though, any ideas welcome